### ISSUE TITLE
[rllib] how to set zero learning rate after checkpoint

### ISSUE THREAD

**Issue by esquires** [ray-project/ray#10241]
## What is your question?

I am trying to change the learning rate after loading in a previous checkpoint but this does not seem to be working for me. Can you help me figure out how to do this? Here is an example:

```python
from pathlib import Path

import gym.envs.classic_control

import ray
from ray import tune


class Cartpole(gym.envs.classic_control.CartPoleEnv):
    def __init__(self, env_config: dict):
        super().__init__()


def main() -> None:
    ray.init()

    first_run = True
    tune_kwargs = {
        "run_or_experiment": "PPO",
        "stop": {"episode_reward_mean": 2000000},
        "config": {"env": Cartpole},
        "checkpoint_freq": 1,
    }

    if first_run:
        tune.run(**tune_kwargs)
    else:
        tune_kwargs['restore'] = \
            Path('~/ray_results/PPO/PPO_Cartpole_0_2020-08-21_09-14-578c2c3za1/checkpoint_4/checkpoint-4')
        tune_kwargs['config']['lr'] = 0.0
        tune.run(**tune_kwargs)


if __name__ == '__main__':
    main()

```

After the first run, the system learns. However, if you set the variable `first_run` to false so that the code goes into the second branch where it loads a checkpoint (you would need to adjust the path on your system) and the learning rate it set to zero), it continues learning. When I look at the console output in the second run, the `cur_lr` is set to 0. Can you help me understand why the system is still learning?

In the example I ran on my system, I let it run through 3 training iterations on the first run so that it ended with a reward of 65. On the second run, it ran from training iteration 4 to 8 and ended with a reward of 275.

*Ray version and other system information (Python version, TensorFlow version, OS):*

* ray version: 0.8.7 (built from source)
* python version: 3.6.9
* OS: ubuntu 18.04
* Pytorch: 1.6.0

**Comment by esquires** [comment]
I ended up just creating this function to set the learning rate to zero:

```python
def make_zero_lr_ckpt(ckpt: str) -> str:
    ckpt_dir = Path(ckpt).parent
    ckpt_dir_no_lr = ckpt_dir.parent / ('.' + ckpt_dir.name + '_no_lr')
    shutil.copytree(ckpt_dir, ckpt_dir_no_lr)

    ckpt_no_lr = ckpt_dir_no_lr / Path(ckpt).name

    extra_data = pickle.load(open(ckpt_no_lr, "rb"))
    worker_dict = pickle.loads(extra_data['worker'])
    worker_dict['state']['default_policy']['_optimizer_variables'][0]['param_groups'][0]['lr'] = 0.0
    extra_data['worker'] = pickle.dumps(worker_dict)
    pickle.dump(extra_data, open(ckpt_no_lr, 'wb'))
    return str(ckpt_no_lr)


```
